test/cluster/test_alternator_proxy_protocol.py

```diff
@@ -0,0 +1,318 @@
+    'alternator_https_port': ALTERNATOR_HTTPS_PORT,
+    'alternator_port_proxy_protocol': ALTERNATOR_PROXY_PORT,
+    'alternator_https_port_proxy_protocol': ALTERNATOR_HTTPS_PROXY_PORT,
+    'alternator_write_isolation': 'only_rmw_uses_lwt',
+    'alternator_enforce_authorization': False,
+    'alternator_encryption_options': {
+        'certificate': 'conf/scylla.crt',
+        'keyfile': 'conf/scylla.key',
+    },
+}
+
+
+@pytest.fixture(scope="function")
+async def alternator_proxy_server(manager: ManagerClient):
+    """Fixture that creates a server with Alternator proxy protocol ports enabled."""
+    server = await manager.server_add(config=ALTERNATOR_PROXY_SERVER_CONFIG)
+    yield (server, manager)
+
+
+@pytest.mark.asyncio
+async def test_alternator_proxy_protocol_basic(alternator_proxy_server):
+    """Test that connections through the Alternator proxy protocol port work."""
+    server, _ = alternator_proxy_server
+
+    status_code, response = await send_alternator_request(
+        host=server.ip_addr,
+        port=ALTERNATOR_PROXY_PORT,
+        operation='ListTables',
+        request_body={},
+        proxy_src_addr="203.0.113.42",
+        proxy_src_port=12345
+    )
+
+    assert status_code == 200, f"Expected status 200, got {status_code}: {response}"
+    assert 'TableNames' in response, f"Expected TableNames in response: {response}"
+
+
+@pytest.mark.asyncio
+async def test_alternator_proxy_protocol_multiple_connections(alternator_proxy_server):
+    """Test multiple connections with different source addresses."""
+    server, _ = alternator_proxy_server
+
+    test_clients = [
+        ("203.0.113.101", 10001),
+        ("203.0.113.102", 10002),
+        ("203.0.113.103", 10003),
+    ]
+
+    for fake_src_addr, fake_src_port in test_clients:
+        status_code, response = await send_alternator_request(
+            host=server.ip_addr,
+            port=ALTERNATOR_PROXY_PORT,
+            operation='ListTables',
+            request_body={},
+            proxy_src_addr=fake_src_addr,
+            proxy_src_port=fake_src_port
+        )
+        assert status_code == 200, f"Expected status 200 for {fake_src_addr}, got {status_code}: {response}"
+        assert 'TableNames' in response, f"Expected TableNames in response for {fake_src_addr}: {response}"
+
+
+@pytest.mark.asyncio
+async def test_alternator_proxy_protocol_ssl_basic(alternator_proxy_server):
+    """Test proxy protocol with TLS encryption."""
+    server, _ = alternator_proxy_server
+
+    status_code, response = await send_alternator_request(
+        host=server.ip_addr,
+        port=ALTERNATOR_HTTPS_PROXY_PORT,
+        operation='ListTables',
+        request_body={},
+        use_ssl=True,
+        proxy_src_addr="203.0.113.50",
+        proxy_src_port=55555
+    )
+
+    assert status_code == 200, f"Expected status 200, got {status_code}: {response}"
+    assert 'TableNames' in response, f"Expected TableNames in response: {response}"
```

Comment:
Again, I think it will be good to verify that your test actually managed to send an ssl request, and to "fake" the given source address and port.